pandas/tests/plotting/test_frame.py

```diff
@@ -588,17 +588,13 @@ def test_subplots_layout(self):
     @pytest.mark.slow
     def test_subplots_warnings(self):
         # GH 9464
-        warnings.simplefilter('error')
-        try:
+        with pytest.warns(Warning):
```

Comment:
Asserting that a warning is raised is too general. Currently this passes, but no warning related to matplotlib are raise (at least locally for me), instead I get,
```
The list of emitted warnings is: 
[DeprecationWarning("Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working"),
 DeprecationWarning("Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working")].
```

Just FYI, it's an unrelated issue.

Reply:
use tm.assert_produces_warning

Reply:
Thanks for the review!

Sorry, I misread the code, it meant to check that no warning are raised, please ignore my comment above. Fixed it now.